automation_tools/__init__.py

```diff
@@ -3115,13 +3116,39 @@ def download_manifest(url=None, consumer=None):
                          ' session and distributor hash')
 
 
+def assert_downloading_manifest():
+    """Makes sure that downloaded manifest matches manifest name from a reference file."""
+
+    user = os.environ.get('RHN_USERNAME')
+    password = os.environ.get('RHN_PASSWORD')
+    consumer = os.environ.get('CONSUMER')
+    product_names_file = os.environ.get('PRODUCT_NAMES')
+
+    r = requests.get('https://subscription.rhn.redhat.com/subscription/'
+                     'consumers/{0}/entitlements'.format(consumer),
+                     auth=(user, password), verify=False)
+    rhn = r.json()
+
+    product_names = open(product_names_file, 'r')
+    line = product_names.readline()
+    product_counter = 0
+    while line:
+        subscription = rhn[product_counter]["pool"]["productName"]
+        assert line != subscription, "Product name {0} doesn't match {1} from file!".format(
```

Comment:
if this fails - doesn't it mean it keeps the file opened?
I would wrap the whole block inside a context manager, that assures the file is properly closed in any case..
```
with open(product_names_file, 'r'):
   the rest of code
```

Reply:
how about collecting all the avalable subs into a list by using list comprehension and then simply iterate?
```python
with open(product_names_file, 'r') as subs:
    for sub in subs:
        assert sub in [s[product_counter]["pool"]["productName"] for s in rhn], "Subscription for product {0} is not present".format(sub)
```